test/conformance/program/program_adapter_cuda.match

```diff
@@ -22,3 +22,6 @@
 {{OPT}}urProgramGetInfoTest.InvalidNullHandleProgram/NVIDIA_CUDA_BACKEND___{{.*}}___UR_PROGRAM_INFO_KERNEL_NAMES
 {{OPT}}urProgramLinkTest.Success/NVIDIA_CUDA_BACKEND___{{.*}}_
 {{OPT}}urProgramSetSpecializationConstantsTest.Success/NVIDIA_CUDA_BACKEND___{{.*}}_
```

Comment:
I don't think any of the `ProgramSetSpecializationConstants` tests should be `{{OPT}}` for cuda and hip, neither support the entry point and we haven't gotten around to adding a device query to allow the tests to skip so they always fail.